src/dbnode/client/session.go

```diff
@@ -2151,6 +2155,12 @@ func (s *session) streamBlocksMetadataFromPeers(
 					atomic.AddInt32(&success, 1)
 					return
 				}
+
+				// Prevent the loop from spinning too aggressively if
```

Comment:
how about sleeping for a longer duration only if the specialised error condition is triggered